api/src/utils/logger.ts

```diff
@@ -42,41 +33,69 @@ const getLoggerTransportTypes = (): string[] => {
  *
  * log.debug({ label: 'functionName', message: 'A debug message!:', debugInfo1, debugInfo2 })
  *
- * ...etc
- *
  * Example Output:
  *
- * [15-09-2019 14:44:30] [info] (class-or-file-name): A basic log message!
- *
- * [15-09-2019 14:44:30] [info] (class-or-file-name): functionName - A message with a label!
- *
- * [02-12-2019 14:45:02] [error] (class-or-file-name): functionName - An error message!
  * {
- *   error: 404 Not Found
+ *   timestamp: '2025-02-04 14:05:24',
+ *   level: 'debug',
+ *   message: {
+ *     logger: 'class-or-file-name',
+ *     label: 'functionName',
+ *     message: 'An error message!:',
+ *     error: {...}
+ *   }
  * }
  *
- * [02-12-2019 14:46:15] [error] (class-or-file-name): functionName - A debug message!
- * {
- *   debugInfo1: 'someDebugInfo1'
- * }
- * {
- *   debugInfo2: 'someDebugInfo2'
- * }
- *
- * ...etc
- *
- * Environment Variables:
- *
- * LOG_LEVEL - Defines the level of logging that the logger will output to the console. (default: debug)
- *
- * LOG_LEVEL_FILE - Defines the level of logging that the logger will output to persistent log files. (default: debug)
+ * @param {string} logLabel common label for the instance of the logger.
+ * @returns
+ */
+export const getLogger = (logLabel: string) => {
+  const logger = _getLogger(DEFAULT_LOGGER);
+
+  return {
+    info: (params: CustomWinstonLoggerParams) => logger.info({ logger: logLabel, ...params }),
+    warn: (params: CustomWinstonLoggerParams) => logger.warn({ logger: logLabel, ...params }),
+    error: (params: CustomWinstonLoggerParams) => logger.error({ logger: logLabel, ...params }),
+    debug: (params: CustomWinstonLoggerParams) => logger.debug({ logger: logLabel, ...params }),
+    silly: (params: CustomWinstonLoggerParams) => logger.silly({ logger: logLabel, ...params })
+  };
+};
+
+/**
+ * Get the transport types to use for the logger.
  *
- * Valid logging level values (from least logging to most logging) - silent, error, warn, info, debug, silly
+ * @return {*}  {string[]}
+ */
+const getLoggerTransportTypes = (): string[] => {
+  const transportTypes = [];
+
+  // Do not output logs to file when running unit tests
+  // Note: Both lifecycle events are needed to prevent log files ie: `npm run test` or `npm run test-watch`
+  if (process.env.npm_lifecycle_event !== 'test' && process.env.npm_lifecycle_event !== 'test-watch') {
+    transportTypes.push('file');
+  }
+
+  if (process.env.NODE_ENV !== 'production') {
+    transportTypes.push('console');
+  }
+
+  return transportTypes;
+};
+
+/**
+ * Get or create a singleton logger instance.
  *
- * @param {string} logLabel common label for the instance of the logger.
+ * @param {string} loggerName The name of the logger instance.
  * @returns
  */
-export const getLogger = function (logLabel: string) {
+export const _getLogger = function (loggerName: string) {
```

Comment:
nit: We could use a different name for this function that is a bit more explicit. 
ie: `_getOrCreateLoggerSingleton`